### PR TITLE
Ab1800 filename validation

### PR DESCRIPTION
**Pull request by StuartHarris**
[Ticket 1800](https://dev.azure.com/futurenhs/FutureNHS/_boards/board/t/FutureNHS%20Team/Stories/?workitem=1800)

see issue #387 

## Acceptance Criteria

- [x] file_name must be between 5 and 255 characters long
- [x] filename contains characters that are not alphanumeric, space, period, hyphen or underscore
- [x] filename does not have a correctly formatted extension
- [x] filename extension is in whitelist (for now these are: doc, docx, pdf, eps, xls, xslx, ppt, pptx, jpg, jpeg, gif, png, bmp, webp, svg)
- [x] `Content-Type` header is correct for the filename extension

## Pre-review checklist:

- [x] Tests

- [ ] Documentation

## Testing information

- Is there any special setup required?

- Test plan?

## Test:

- [x] Tested locally

### CODE REVIEW

File: workspace-service/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+target
+.env
```
Comment (frigus02) [inline workspace-service/.gitignore:2]: I would have expected those to be covered by the [root level .gitignore](https://github.com/FutureNHS/futurenhs-platform/blob/master/.gitignore). I guess it doesn't hurt to have a local one, too, though.

Comment (StuartHarris) [reply workspace-service/.gitignore:2]: They are. But if you open just the workspace service folder in vscode, it doesn't see the higher level

Comment (StuartHarris) [reply workspace-service/.gitignore:2]: so searches through `./target` etc

Comment (StuartHarris) [reply workspace-service/.gitignore:2]: nice to have each service's root be independent I think

File: workspace-service/src/graphql/files.rs [hunk 2]
```diff
@@ -2,8 +2,36 @@ use super::azure;
...
+            (jpg)|
+            (pdf)|
+            (png)|
+            (ppt)|
+            (pptx)|
+            (svg)|
+            (txt)|
+            (webp)|
+            (xls)|
+            (xslx)
```
Comment (frigus02) [inline workspace-service/src/graphql/files.rs:30]: We might want to add the OpenDocument extensions:
- .odt for text documents
- .ods for spreadsheets
- .odp for presentations

Comment (StuartHarris) [reply workspace-service/src/graphql/files.rs:34]: Thanks! I added these

File: workspace-service/src/graphql/files.rs [hunk 3]
```diff
@@ -2,8 +2,40 @@ use super::azure;
...
+use mime_db::extensions2;
+use regex::Regex;
 use url::Url;
 use uuid::Uuid;
+use validator::{Validate, ValidationError};
+
+lazy_static! {
+    static ref ALLOWED_FILENAME_CHARS: Regex = Regex::new(r"^[\w\s\.-]+$").expect("bad regex");
+    static ref ALLOWED_EXTENSIONS: Regex = Regex::new(
+        r"(?x)^[\w\s\.-]+(
```
Comment (StuartHarris) [inline workspace-service/src/graphql/files.rs:15]: enforce a dot before the extension

**Review by frigus02 — APPROVED** [review]

**Pull request merged**